### PR TITLE
feat(rum-core): add serverUrlPrefix config option

### PR DESCRIPTION
**Pull request by alexbode**
Change hardcoded serverUrlPrefix to passable parameter to evade ad blockers

fixes issue #1078

### CODE REVIEW

**Review by vigneshshanmugam — COMMENTED** [review]
Looks good, can you add test here - https://github.com/elastic/apm-agent-rum-js/blob/master/packages/rum-core/test/common/apm-server.spec.js and also add docs on the new flag - https://github.com/elastic/apm-agent-rum-js/blob/master/docs/configuration.asciidoc

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -21,6 +21,21 @@ Your Elastic APM service name.
 
 The URL used to make requests to the APM Server.
 
+[float]
+[[server-prefix-url]]
+=== `serverPrefixUrl`
```
Comment (vigneshshanmugam) [inline docs/configuration.asciidoc:26]: need to be `serverUrlPrefix`

File: docs/configuration.asciidoc [hunk 2]
```diff
@@ -21,6 +21,21 @@ Your Elastic APM service name.
...
+
+* *Type:* String
```
Comment (vigneshshanmugam) [inline docs/configuration.asciidoc:28]: @bmorelli25 Can you help with the review on the docs?

File: docs/configuration.asciidoc [hunk 3]
```diff
@@ -21,6 +21,21 @@ Your Elastic APM service name.
...
+* *Default* `/intake/v${apiVersion}/rum/events`
+
+The server prefix url used to make requests to the APM Server. Some ad blockers block outgoing requests
+from the browser if the default server prefix url is being used. Using a custom server prefix url can be used to
+evade the ad blocker.
+
+NOTE: If the default server prefix url is overwritten, the reverse proxy that sits between the
```
Comment (vigneshshanmugam) [inline docs/configuration.asciidoc:35]: Nice little note :)

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

File: docs/configuration.asciidoc (hunk 1 above)
Comment (bmorelli25) [reply docs/configuration.asciidoc:26]:
```suggestion
=== `serverUrlPrefix`
```

File: docs/configuration.asciidoc [hunk 4]
```diff
@@ -21,6 +21,21 @@ Your Elastic APM service name.
 
 The URL used to make requests to the APM Server.
 
+[float]
+[[server-prefix-url]]
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:25]:
```suggestion
[[server-url-prefix]]
```

File: docs/configuration.asciidoc [hunk 5]
```diff
@@ -21,6 +21,21 @@ Your Elastic APM service name.
...
+* *Type:* String
+* *Default* `/intake/v${apiVersion}/rum/events`
+
+The server prefix url used to make requests to the APM Server. Some ad blockers block outgoing requests
+from the browser if the default server prefix url is being used. Using a custom server prefix url can be used to
+evade the ad blocker.
+
+NOTE: If the default server prefix url is overwritten, the reverse proxy that sits between the 
+APM Server and the rum agent must reroute traffic to `/intake/v${apiVersion}/rum/events`
+so the APM Server knows what intake api to use.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:37]:
```suggestion
The server prefix URL used to make requests to the APM Server. Some ad blockers block outgoing requests
from the browser if the default server prefix URL is being used. Using a custom server prefix URL can be used to
evade the ad blocker.

NOTE: If the default server prefix URL is overwritten, the reverse proxy that sits between the 
APM Server and the rum agent must reroute traffic to `/intake/v${apiVersion}/rum/events`
so the APM Server knows what intake API to use.
```

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by bmorelli25** [comment]
jenkins test this please

**Review by bmorelli25 — APPROVED** [review]
Thanks, @alexbode !

**Comment by vigneshshanmugam** [comment]
@alexbode Can you rebase your commits with `feat(rum-core): add serverUrlPrefix config option` just to make sure the tests get picked up and lint pass.

**Pull request merged**